### PR TITLE
Catch unauthorized errors during sync

### PR DESCRIPTION
**Pull request by navarone-feekery**
Catch errors when connector is unauthorized to access a content index during sync and provide a human-readable/actionable error message.

Previous to this the error output would be:

```
Sync failure: AuthorizationException: AuthorizationException(403, 'security_exception', 'action [indices:data/read/search] is unauthorized for API key id [<API key id>] of user [<username>] on indices [<index-name>], this action is granted by the index privileges [read,all]').
```

### CODE REVIEW

File: connectors/sync_job_runner.py [hunk 1]
```diff
@@ -192,6 +195,15 @@ async def execute(self):
             await self._sync_done(sync_status=JobStatus.SUSPENDED)
         except ConnectorJobCanceledError:
             await self._sync_done(sync_status=JobStatus.CANCELED)
+        except ElasticAuthorizationException as e:
+            index_name = (
+                self.connector.acl_index_name
+                if job_type == JobType.ACCESS_CONTROL
+                else self.connector.index_name
+            )
```
Comment (wangch079) [inline connectors/sync_job_runner.py:203]: You don't need this. You can always use `self.sync_job.index_name`:

https://github.com/elastic/connectors/blob/2d5df6a3ac11a6b6124886c980f406ed7ce1d052/connectors/protocol/connectors.py#L1002-L1007

**Review by wangch079 — DISMISSED** [review]
LGTM, left one comment

File: connectors/sync_job_runner.py (hunk 1 above)
Comment (navarone-feekery) [reply connectors/sync_job_runner.py]: Oh neat! I've changed it to use this. Thank you.

**Comment by navarone-feekery** [comment]
@wangch079 can you re-review please? Adding changes set your review to stale.

**Review by timgrein — APPROVED** [review]
Nice change 🚀

**Pull request merged**